src/main/java/com/github/tomakehurst/wiremock/http/trafficlistener/ConsoleNotifyingWiremockNetworkTrafficListener.java

```diff
@@ -16,44 +16,63 @@
 package com.github.tomakehurst.wiremock.http.trafficlistener;
 
 import com.github.tomakehurst.wiremock.common.ConsoleNotifier;
+import com.github.tomakehurst.wiremock.common.Notifier;
+
 import java.net.Socket;
 import java.nio.ByteBuffer;
 import java.nio.charset.CharacterCodingException;
 import java.nio.charset.Charset;
 import java.nio.charset.CharsetDecoder;
+import java.nio.charset.StandardCharsets;
 
 public class ConsoleNotifyingWiremockNetworkTrafficListener
     implements WiremockNetworkTrafficListener {
-  private static final ConsoleNotifier CONSOLE_NOTIFIER = new ConsoleNotifier(true);
+  private static final ConsoleNotifier DEFAULT_CONSOLE_NOTIFIER = new ConsoleNotifier(true);
+  private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
+
+  private final Notifier notifier;
+  private final Charset charset;
+  private final CharsetDecoder charsetDecoder;
+
+  ConsoleNotifyingWiremockNetworkTrafficListener(Notifier notifier, Charset charset) {
```

Comment:
Maybe public too? Can be done later if needed

Reply:
It's been a little bit, but this choice was made around the fact that this type is named `ConsoleNotifying*`.  It seemed off to allow for instantiating it with a custom notify that could notify to places other than the console.  So per that consideration and Effective Java [Item 13](https://github.com/HugoMatilla/Effective-JAVA-Summary#13-minimize-the-accessibility-of-classes-and-members), I'm intentionally minimizing accessibility.

I do think there's potentially an opportunity here to create a simple or [skeletal](https://github.com/HugoMatilla/Effective-JAVA-Summary#18-prefer-interfaces-to-abstract-classes)  type that would expose the ability to create a `WiremockNetworkTrafficListener` with an arbitrary notifier charset combo.

I'd probably have that be structured as:

```
public final class WiremockNetworkTrafficListeners {
  private WiremockNetworkTrafficListeners() {}

  public static create(Notifier notifier, CharSet charset) {
    return new SimpleWiremockNetworkTrafficListener(notifier, charset);
  }

  public static createConsoleNotifying() {
    return new SimpleWiremockNetworkTrafficListener(DEFAULT_CONSOLE_NOTIFIER, DEFAULT_CHARSET)
  }
  
  public static createConsoleNotifying(CharSet charset) {
    return new SimpleWiremockNetworkTrafficListener(DEFAULT_CONSOLE_NOTIFIER, charset)
  }  

  public static create()
}

class SimpleWiremockNetworkTrafficListener implements WiremockNetworkTrafficListener {
  // Basically all the implementation that currently exists in ConsoleNotifying
}
```

In Wiremock 2 you'd probably keep a shim `ConsoleNotifyingNetworkTrafficListener` type, but in 3 I'd drop it altogether for the `WiremockNetworkTrafficListeners` factory pattern.

Let me know what you think.  But in any case, as long as it's okay with you let's do it as part of a separate PR?

Reply:
I think it would be a good improvement for WireMock's API extensibility!
If you are open to submitting a separate PR for that, indeed it would be the best approach